### PR TITLE
ddl: use a valid `ctx` instead of `context.Background`

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:
When DDL exits, some related functions may not exit in time

### What is changed and how it works?

What's Changed: 

- Use a valid `ctx` instead of `context.Background`

- Use `context` instead of `quitCh` to unify the exit logic.

- Remove useless code.

How it Works:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

-

### CODE REVIEW

File: ddl/util/syncer_test.go [hunk 1]
```diff
@@ -221,14 +221,14 @@ func TestSyncerSimple(t *testing.T) {
 	}
 	checkRespKV(t, 0, ttlKey, "", resp.Kvs...)
 
-	// for RemoveSelfVersionPath
+	// for Close
```
Comment (tangenta) [inline ddl/util/syncer_test.go:224]: Does it mean "test for SchemaSyncer.Close()"?

Comment (zimulala) [reply ddl/util/syncer_test.go:224]: Yes. This test is `TestSyncerSimple`, so I don't add `SchemaSyncer`.

**Comment by zimulala** [comment]
PTAL @tangenta @wjhuang2016

**Comment by zimulala** [comment]
A test failed:
```
[2020-07-07T06:31:59.587Z] FAIL: sort_test.go:26: testSuite.TestSortInDisk

[2020-07-07T06:31:59.587Z] 
[2020-07-07T06:31:59.587Z] sort_test.go:59:
[2020-07-07T06:31:59.587Z]     c.Assert(tk.Se.GetSessionVars().StmtCtx.DiskTracker.MaxConsumed(), Greater, int64(0))
[2020-07-07T06:31:59.587Z] ... compare_one int64 = 0
[2020-07-07T06:31:59.587Z] ... compare_two int64 = 0
```

**Comment by zimulala** [comment]
A test failed:
```
[2020-07-07T06:46:39.334Z] FAIL: conn_test.go:306: ConnTestSuite.TestDispatchClientProtocol41

[2020-07-07T06:46:39.334Z] 
[2020-07-07T06:46:39.335Z] conn_test.go:406:
[2020-07-07T06:46:39.335Z]     ts.testDispatch(c, inputs, mysql.ClientProtocol41)
[2020-07-07T06:46:39.335Z] conn_test.go:434:
[2020-07-07T06:46:39.335Z]     c.Assert(err, IsNil)
[2020-07-07T06:46:39.335Z] ... value *errors.withStack = listen tcp 0.0.0.0:4005: bind: address already in use ("listen tcp 0.0.0.0:4005: bind: address already in use")
```

**Pull request merged**